dvc/remote/base.py

```diff
@@ -485,7 +489,10 @@ def download(self, from_info, to_info, name=None, no_progress_bar=False):
             logger.exception(msg.format(from_info, to_info))
             return 1  # 1 fail
 
-        move(tmp_file, fspath_py35(to_info))
+        if self.repo:
+            self.repo.cache.local.move(PathInfo(tmp_file), to_info)
+        else:
+            move(tmp_file, fspath_py35(to_info))
```

Comment:
This looks fishy. Why can't we ask cache config here and pass appropriate mode to `move()`?